tensorboard/plugins/graph/tf_graph_common/op.ts

```diff
@@ -14,435 +14,438 @@ limitations under the License.
+      'TensorArrayScatterV3',
+      'TensorArraySizeV3',
+      'TensorArraySplitV3',
+      'TensorArrayV3',
+      'TensorArrayWriteV3',
+      'Tile',
+      'TopKV2',
+      'Transpose',
+      'TruncateDiv',
+      'TruncateMod',
+      'TruncatedNormal',
+      'Unpack',
+      'UnsortedSegmentMax',
+      'UnsortedSegmentMin',
+      'UnsortedSegmentProd',
+      'UnsortedSegmentSum',
+      'VarIsInitializedOp',
+      'VariableShape',
+      'While',
+      'XlaDynamicUpdateSlice',
+      'XlaHostCompute',
+      'XlaIf',
+      'XlaRecv',
+      'XlaReduceWindow',
+      'XlaSend',
+      'XlaSort',
+      'XlaWhile',
+      'ZerosLike',
 
-  /**
-   * Returns true if the node's inferred device is not the TPU.
-   * Note that this is only a best-effort check.
-   */
-  export function isNotTpuOp(opDevice: string): boolean {
-    if (opDevice.toLowerCase().search('cpu:') != -1) {
-      return true;
-    }
-    if (opDevice.toLowerCase().search('gpu:') != -1) {
-      return true;
-    }
-    return (opDevice.toLowerCase().search('tpu') == -1);
-  }
+      // Ops below are manually whitelisted and should not be evaluated for
+      // compatibility for various reasons.
 
-  /**
-   * Returns true if OpNode graph object represents a
-   * Tensorflow operation that is valid for the TPU.
-   */
-  export function opValid(opNode: OpNode): boolean {
-    // Function library nodes are generally for internal use.
-    if (opNode.name.search(FUNCTION_LIBRARY_NODE_PREFIX) == 0) {
-      return true;
-    }
-    // Nodes that lack op types should be ignored.
-    if (!opNode.op) {
-      return true;
-    }
-    // If assigned a device that is not TPU-related assume op is valid.
-    if (opNode.device && isNotTpuOp(opNode.device)) {
-      return true;
+      // Control flow ops.
+      'Enter',
+      'Exit',
+      'LoopCond',
+      'Merge',
+      'NextIteration',
+      'Switch',
+      // Ops below are inserted by the compiler.
+      '_Arg',
+      '_ParallelConcatUpdate',
+      '_Retval',
+      '_TPUCompile',
+      '_TPUExecute',
+      // Distributed TPU ops.
+      'TPUCompilationResult',
+      'TPUReplicatedInput',
+      'TPUReplicatedOutput',
+      'TPUReplicateMetadata',
+      // Checkpointing ops.
+      'MergeV2Checkpoints',
+      'RestoreV2',
+      'SaveV2',
+      // Miscellaneous CPU ops.
+      'Abort',
+      'Assert',
+      'Assign',
+      'Placeholder',
+      'PlaceholderV2',
+      'ShardedFilename',
+      'StringJoin',
+      'Variable',
+      'VariableV2',
+      'VarHandleOp',
+      // Summary ops.
+      'AudioSummary',
+      'AudioSummaryV2',
+      'DebugNumericSummary',
+      'HistogramSummary',
+      'ImageSummary',
+      'MergeSummary',
+      'ScalarSummary',
+      'StatsAggregatorSummary',
+    ];
+
+    /**
+     * Returns true if the node's inferred device is not the TPU.
+     * Note that this is only a best-effort check.
+     */
+    isNotTpuOp(opDevice: string): boolean {
```

Comment:
please prefix this with `private`. Though it used to be exported, it is not used anywhere else.